src/CodeGen.js

```diff
@@ -70,42 +71,44 @@ ${dimensionMappingsSection()}
     return `// Internal state
 bool lookups_initialized = false;
 
-void initConstants() {
-  // Initialize constants.
-${section(Model.constVars())}
-
+void initLookups() {
   // Initialize lookups.
-if (!lookups_initialized) {
-${section(Model.lookupVars())}
-${section(Model.dataVars())}
-  lookups_initialized = true;
-}
-}
-void initLevels() {
-  // Initialize variables with initialization values, such as levels, and the variables they depend on.
-  _time = _initial_time;
-${section(Model.initVars())}
+  if (!lookups_initialized) {
+    ${section(Model.lookupVars())}
+    ${section(Model.dataVars())}
+    lookups_initialized = true;
+  }
 }
 
+${chunkedFunctions('initConstants', Model.constVars(),
+'  // Initialize constants.',
+'  initLookups();'
+)}
+
+${chunkedFunctions('initLevels', Model.initVars(), `
```

Comment:
The chunkedFunctions() is a bit more dynamic (swizzling?) than I'm used to so, take this for what you will, but I noticed that you call it with three params sometimes and four others.  I assume that means the postStep has a default value/action, but just in case...

Reply:
Yep, that's correct, `postStep` is essentially an optional argument.  If it's not included, it becomes undefined and the `chunkedFunctions` function has an `if (postStep)` check that means "only include this string if it's defined".